File: python_implementation/src/application/use_cases/tasks/create_task_use_case.py
from src.domain.abstractions.repositories.abstract_task_repository import AbstractTaskRepository
from src.domain.entities.task_entity import TaskEntity
from src.domain.enums.task_enums import TaskStatus
from src.application.dtos.task_dtos import CreateTaskRequest


class CreateTaskUseCase:
    """Use case for creating a new task"""
    
    def __init__(self, task_repository: AbstractTaskRepository):
        self.task_repository = task_repository
    
    def execute(self, request: CreateTaskRequest) -> TaskEntity:
        task_entity = TaskEntity(
            title=request.title,
            status=TaskStatus.OPEN
        )
        return self.task_repository.create(task_entity)

File: java_implementation/src/main/java/com/example/taskapi/application/usecases/CreateTaskUseCase.java
package com.example.taskapi.application.usecases;

import com.example.taskapi.application.dtos.CreateTaskRequest;
import com.example.taskapi.domain.enums.TaskStatus;
import com.example.taskapi.domain.model.Task;
import com.example.taskapi.domain.repository.TaskRepositoryPort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CreateTaskUseCase {
    private final TaskRepositoryPort taskRepository;

    public Task execute(CreateTaskRequest request) {
        Task task = Task.builder()
                .title(request.getTitle())
                .status(TaskStatus.OPEN)
                .build();
        return taskRepository.create(task);
    }
}
